### ISSUE TITLE
Using tracing instrumentation to measure call timings 

### ISSUE THREAD

**Issue by Seba-T** [kanidm/kanidm#2830]
### Is your feature request related to a problem? Please describe

I was looking into tracing [instrumentation](https://docs.rs/tracing/0.1.38/tracing/attr.instrument.html) to understand how we could use it to get call timings, only to conclude that perhaps we'd need an [additional crate](https://stackoverflow.com/a/74711569) to do that. So my question is: is it worth it to add yet another dependency or should we keep it the way it is?

### Describe the solution you'd like

Avoiding Kani getting even chonkier, but what do I know anyway.

### Describe alternatives you've considered

Going for a walk.

**Comment by yaleman** [comment]
Tracing instrumentation and OpenTelemetry integration's already built in. You can export traces to an OTLP endpoint by configuring the [otel_grpc_url](https://kanidm.github.io/kanidm/master/monitoring_the_platform.html?highlight=monito#opentelemetry-tracing) 😄 


And if you collect said traces you can get handy timing graphs like this:

<img width="1133" alt="Screenshot 2024-06-09 at 08 39 00" src="https://github.com/kanidm/kanidm/assets/168188/1bf4f6d8-5060-4e7d-915d-55007171835c">

I was playing with a very simple test project over in https://github.com/yaleman/test-tracing-instrument which just dumps to stdout if you want to handle that mess, too.

**Comment by Seba-T** [comment]
Ohhhh so opentelemetry became a thing in kani eventually?!? that is very good to know, I think I was stuck to 6 months ago when it was still being debated :p.
So if I understand this correctly kani sends telemetry data to the provided endpoint? Which makes me think I'd need an endpoint or something 🤔

**Comment by yaleman** [comment]
Yes, you'd need to run a trace collector to accept it, there's example code for setting it up in https://github.com/kanidm/kanidm/tree/master/scripts/otel

**Comment by Seba-T** [comment]
Nice! I was worried I had to set up everything from scratch myself. We will need to add this to orca though, or am I flipping @Firstyear? More specifically I mean we will need to get rid of all the stats collecting part in orca and move it to something similar to the otel container.

**Comment by Firstyear** [comment]
You'd have both as options.
